src/components/transport_manager/src/usb/libusb/usb_handler.cc

```diff
@@ -320,7 +324,10 @@ void UsbHandler::Thread() {
   SDL_TRACE("enter");
   int completed = 0;
   while (!shutdown_requested_) {
-    libusb_handle_events_completed(libusb_context_, &completed);
+    // all zero timeval struct for non-blocking mode
+    struct timeval max_wait_time = {0, 0};
+    libusb_handle_events_timeout_completed(
```

Comment:
@pestOO See 8e59950a3a3aa8cc31fc5731eb3e2fc72d6421a1 According to [Using an event handling thread](http://libusb.sourceforge.net/api-1.0/group__asyncio.html) the blocking `  libusb_handle_events` is unlocked by the call to `libusb_hotplug_deregister_callback`